lib/loader.js

```diff
@@ -41,8 +41,8 @@ module.exports = function (content) {
   this.cacheable()
   var isServer = this.options.target === 'node'
   var loaderContext = this
-  var options = this.vue || this.options.vue || {}
   var query = loaderUtils.parseQuery(this.query)
+  var options = this.options.vue || query || {}
```

Comment:
I think we should

1. keep `this.vue` for current usage compatibility
2. prioritize query first

So perhaps `query || this.vue || this.options.vue || {}`